.changeset/khaki-books-rescue.md

```diff
@@ -0,0 +1,33 @@
+---
+'@cloudflare/next-on-pages': minor
+---
+
+add new `utils` sum-module with `getRequestCfProperties` and `getRequestExecutionContext`
+
+add a new `utils` module that will provide utilities that will allow developer to utilize
+better next-on-pages and the cloudflare platform.
+
+alongside the `utils` two functions have been introduced `getRequestCfProperties` and
+`getRequestExecutionContext`, the former provides to developers the `cf` object from
+the request, whilst the latter the execution context object (`ctx`), both results
+are properly typed (with the types coming from `@cloudflare/workers-types`).
+
+Note that naturally both only work on the server, they throw an exception if run on the client,
+where there is not request to get the objects from.
+
+Usage example:
+
+```ts
+import {
+	getRequestCfProperties,
+	getRequestExecutionContext,
```

Comment:
the `Properties` part of `getRequestCfProperties` was just to match the workers types type: `IncomingRequestCfProperties`, since it's included there I felt as it could make things clearer to have it here as well?

regarding `getRequestExecutionContext` I can't really remember for sure but I _think_ that I included `Request` so that that the two methods have a very similar format making their use a bit easier/streamlines 🤔 

I don't mind changing the names, but what do you think of the above?